src/index.js

```diff
@@ -1,123 +1,117 @@
 const TESTING = global.COMB !== undefined;
-
-if (!TESTING)
+if (!TESTING) {
   COMB = require('@holo-host/comb').COMB;
+}
 
 const { EventEmitter } = require('events');
 
-function makeUrlAbsolute (url) {
+function makeUrlAbsolute(url) {
   return new URL(url, window.location).href
 }
 
-class Connection extends EventEmitter {
-
-  constructor(url, signalCb, opts) {
+  // Child Msg Bus Alert Types:
+  // - `sign-in` - emitted when the user completes a successful sign-in
+  // - `sign-up` - emitted when the user completes a successful sign-up
+  // - `sign-out` - emitted when the user competes a successful sign-out
+  // - `canceled` - emitted when the user purposefully exits sign-in/up
+  // - `signal` - emitted when a signal is passed from chaperone
+  // - `available` - emitted when the WebSDK to chaperone and envoy are opened and hosted app is ready for zome calls
+  // - `unavailable` - emitted when the ws WebSDK to chaperone and/or envoy is closed
+  // - `unrecoverable-agent-state` - emitted when an unrecoverable error event is passed from chapeone
+
+class WebSdkApi extends EventEmitter {
+  constructor(child) {
     super();
+    this.available = null;
+    this.child = child;
+    child.msg_bus.on("alert", (event, ...args) => this.emit(event));
+    child.msg_bus.on("signal", signal => this.emit('signal', signal));
+    child.msg_bus.on("unrecoverable-agent-state", () => this.emit('unrecoverable-agent-state'));
+    child.msg_bus.on("unavailable", () => this.emit('unavailable'));
+    child.msg_bus.on("available", () => {
+      this.available()
+      this.emit('available')
+    });
+  }
 
+  ready = () => {
+    return new Promise((resolve, reject) => {
+      this.available = resolve
+    });
+  }
+
+  static connect = async ({ chaperoneUrl, authFormCustomization } = {}) => {
     const hostname = window.location.hostname;
-    this.chaperone_url = new URL(url || `http://${hostname}:24273`);
-    if (opts !== undefined) {
-      if (opts.logo_url !== undefined) {
-        this.chaperone_url.searchParams.set("logo_url", makeUrlAbsolute(opts.logo_url))
+    const chaperone_url = new URL(chaperoneUrl || `http://${hostname}:24273`);
```

Comment:
```suggestion
    const final_chaperone_url = new URL(chaperoneUrl || `http://${hostname}:24273`);
```
having distinct variables `chaperone_url` and `chaperoneUrl` is asking for trouble